src/applications/vaos/tests/new-appointment/components/ConfirmationPage/ConfirmationRequestInfo.unit.spec.jsx

```diff
@@ -189,30 +197,39 @@ describe('VAOS <ConfirmationRequestInfo>', () => {
     };
     const pageTitle = 'Your appointment request has been submitted';
 
-    const tree = mount(
+    const screen = renderWithStoreAndRouter(
       <ConfirmationRequestInfo
         data={data}
         facilityDetails={facilityDetails}
         pageTitle={pageTitle}
       />,
+      {
+        initialState,
+        reducers,
+      },
     );
 
-    tree
-      .find('AdditionalInfo')
-      .find('button')
-      .simulate('click');
-    tree.setProps();
+    // Simulate user clicking the "Show more" button
+    const button = screen.getByRole('button', { name: 'Show more' });
+    userEvent.click(button);
 
-    const text = tree.text();
+    // Button text should change to "Show less"
+    screen.getByRole('button', { name: 'Show less' });
```

Comment:
When I've tested this in other places, I try to do an `await findByText` call before checking for any text to make sure the UI has time to reveal the text, since it might not be fully synchronous and we want to make sure we don't start checking for text before the code has had time to render it.

https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/vaos/tests/list/pending-appointments.unit.spec.jsx#L281

Reply:
Just following suggested guidelines here [https://kentcdodds.com/blog/common-mistakes-with-react-testing-library](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library)

Reply:
Which guideline in there do you mean?

Reply:
https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-the-wrong-query

Reply:
Oh, gotcha. You can still query by role, but use the helper that will wait for the element to appear: `await findByRole`.

Reply:
I misunderstood part of the comment. I changed the test to `await findByRole`. The guideline was about using get/findByRole when possible.

Reply:
Done.